plantcv/plantcv/spectral_index/spectral_index.py

```diff
@@ -0,0 +1,863 @@
+        red_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 680)
+        nir = (hsi.array_data[:, :, [nir_index]])
+        red = (hsi.array_data[:, :, [red_index]])
+        # Naturally ranges from -2 to 2
+        index_array_raw = nir - red
+        return _package_index(hsi=hsi, raw_index=index_array_raw, method="GDVI")
+    else:
+        fatal_error("Available wavelengths are not suitable for calculating GDVI. Try increasing distance.")
+
+def savi(hsi, distance=20):
+    """Soil Adjusted Vegetation Index [Huete et al. (1988)]
+    The theoretical range for SAVI is [-1.2, 1.2].
+
+    inputs:
+    hsi      = hyperspectral image (PlantCV Spectral_data instance)
+    distance = how lenient to be if the required wavelengths are not available
+
+    Returns:
+    index_array    = Index data as a Spectral_data instance
+    :param hsi: __main__.Spectral_data
+    :param distance: int
+    :return index_array: __main__.Spectral_data
+    """
+    if (float(hsi.max_wavelength) + distance) >= 800 and (float(hsi.min_wavelength) - distance) <= 680:
+        nir_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 800)
+        red_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 680)
+        nir = (hsi.array_data[:, :, [nir_index]])
+        red = (hsi.array_data[:, :, [red_index]])
+        # Naturally ranges from -1.2 to 1.2
+        index_array_raw = (1.5 * (nir - red)) / (red + nir + 0.5)
+        return _package_index(hsi=hsi, raw_index=index_array_raw, method="SAVI")
+    else:
+        fatal_error("Available wavelengths are not suitable for calculating SAVI. Try increasing distance.")
+
+def pri(hsi, distance=20):
+    """Photochemical Reflectance Index (https://doi.org/10.1111/j.1469-8137.1995.tb03064.x)
+    The theoretical range for PRI is [-1.0, 1.0].
+
+    inputs:
+    hsi      = hyperspectral image (PlantCV Spectral_data instance)
+    distance = how lenient to be if the required wavelengths are not available
+
+    Returns:
+    index_array    = Index data as a Spectral_data instance
+    :param hsi: __main__.Spectral_data
+    :param distance: int
+    :return index_array: __main__.Spectral_data
+    """
+    if (float(hsi.max_wavelength) + distance) >= 570 and (float(hsi.min_wavelength) - distance) <= 531:
+        # Obtain index that best approximates 570 and 531 nm bands
+        pri570_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 570)
+        pri531_index = _find_closest(np.array([float(i) for i in hsi.wavelength_dict.keys()]), 531)
+        pri570 = (hsi.array_data[:, :, [pri570_index]])
+        pri531 = (hsi.array_data[:, :, [pri531_index]])
+        # PRI = (R531- R570)/(R531+ R570))
+        denominator = pri531 + pri570
+        # Avoid dividing by zero
+        index_array_raw = np.where(denominator == 0, 0, ((pri531 - pri570) / denominator))
```

Comment:
Looks like in PRI we are avoiding division by zero but we are not doing that in the indices above. Since we are going to handle `nan` outputs later, I think we should just make this a simple equation like the other indices.

Reply:
Sure, I have made the changes.